build/common.sh

```diff
@@ -833,3 +833,43 @@ function kube::release::gcs::publish_latest() {
   echo "+++ gsutil cat ${latest_file_dst}:"
   gsutil cat ${latest_file_dst}
 }
+
+# Publish a new latest.txt, but only if the release we're dealing with
+# is newer than the contents in GCS.
+function kube::release::gcs::publish_latest_official() {
+  local new_version=${KUBE_GCS_LATEST_CONTENTS}
+  local latest_file_dst="gs://${KUBE_GCS_RELEASE_BUCKET}/${KUBE_GCS_LATEST_FILE}"
+
+  local version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)$"
+  [[ ${new_version} =~ ${version_regex} ]] || {
+    echo "!!! publish_latest_official passed bogus value: '${new_version}'" >&2
+    exit 1
+  }
+
+  local version_major="${BASH_REMATCH[1]}"
+  local version_minor="${BASH_REMATCH[2]}"
+  local version_patch="${BASH_REMATCH[3]}"
+
+  local gcs_version
+  gcs_version=$(gsutil cat ${latest_file_dst})
+
+  [[ ${gcs_version} =~ $version_regex ]] || {
+    echo "!!! ${latest_file_dst} contains invalid release version, can't compare: '${gcs_version}'" >&2
+    exit 1
+  }
+
+  local gcs_version_major="${BASH_REMATCH[1]}"
+  local gcs_version_minor="${BASH_REMATCH[2]}"
+  local gcs_version_patch="${BASH_REMATCH[3]}"
+
+  if [[ "${gcs_version_major}" -gt "${version_major}" ]] ||
```

Comment:
I'll fix the logic (it's going to be a painfully long check). Regarding the short version number, I decide to just start enforcing triplets, since we seem to be using them exclusively now.